### PR TITLE
docs: re-organize readme

### PR DESCRIPTION
**Pull request by Narretz**
Hi,

I've taken the liberty to re-organize the readme.

The goal was primarily to make it easier to find the relevant API faster, because not all options are available for name / tag filtering. There was also a bit of duplicated info.

The ToC should help with finding the info quicker, too.

The only things that have changed content-wise is marked in a PR comment.

- table of contents
- move "api" sections to the top
- separate name + tag filter docs, because not all functionality
is available for both
- group general api info together

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -177,6 +205,8 @@ $ npx cypress run --env grep="-hello world"
 $ npx cypress run --env grep="hello; -world"
 ```
 
+Note: An inverted name filter that matches a suite name does not exclude its tests.
```
Comment (Narretz) [inline README.md:208]: This wasn't in the docs before, and it's nice to know that this doesn't (currently) work.

File: README.md [hunk 2]
```diff
@@ -161,14 +172,31 @@ $ npx cypress run --env grep=hello
...
+
 You can pass multiple title substrings to match separating them with `;` character. Each substring is trimmed.
 
 ```shell
 # run all tests with "hello world" or "auth user" in their title
 $ npx cypress run --env grep="hello world; auth user"
 ```
 
-### negative filter
+### Test suites
```
Comment (Narretz) [inline README.md:184]: I added this analogue to the Test Suites section for tags

**Comment by Narretz** [comment]
@admah I've updated the readme with the latest changes. Also added a commit that fixes the version compat error message.

**Comment by admah** [comment]
Closes #138

**Pull request merged**